test/token/ERC20/ERC20Channel.test.js

```diff
@@ -0,0 +1,126 @@
+    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
+    await token.transfer(tokenChannel.address, 30, { from: sender });
+    const hash = await tokenChannel.generateBalanceHash(20);
+    const senderSig = signMsg(hash, privateKeys[sender]);
+    assert.equal(sender,
+      await tokenChannel.getSignerOfBalanceHash(20, senderSig)
+    );
+    const senderHash = await tokenChannel.generateKeccak256(senderSig);
+    const closingSig = signMsg(senderHash, privateKeys[receiver]);
+    await tokenChannel.cooperativeClose(20, senderSig, closingSig, { from: sender });
+    (await token.balanceOf(sender)).should.be.bignumber
+      .equal(80);
+    (await token.balanceOf(receiver)).should.be.bignumber
+      .equal(20);
+  });
+
+  it('create channel and close it from sender with uncooperativeClose', async function () {
+    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
+    await token.transfer(tokenChannel.address, 30, { from: sender });
+    const hash = await tokenChannel.generateBalanceHash(10);
+    const senderSig = signMsg(hash, privateKeys[sender]);
+    assert.equal(sender,
+      await tokenChannel.getSignerOfBalanceHash(10, senderSig)
+    );
+    await tokenChannel.uncooperativeClose(10, { from: sender });
+    const channelInfo = await tokenChannel.getInfo();
+    assert.equal(parseInt(channelInfo[0]), 30);
+    assert.equal(parseInt(channelInfo[1]), latestTime() + duration.days(1));
+    assert.equal(parseInt(channelInfo[2]), 10);
+    await tokenChannel.closeChannel({ from: sender })
+      .should.be.rejectedWith(EVMRevert);
+    await increaseTimeTo(latestTime() + duration.days(2));
+    await tokenChannel.closeChannel({ from: sender });
+    (await token.balanceOf(sender)).should.be.bignumber
+      .equal(90);
+    (await token.balanceOf(receiver)).should.be.bignumber
+      .equal(10);
+  });
+
+  it('create channel and close it from receiver after sender challenge with different balance', async function () {
+    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
+    await token.transfer(tokenChannel.address, 30, { from: sender });
+    const hash = await tokenChannel.generateBalanceHash(20);
+    const senderSig = signMsg(hash, privateKeys[sender]);
+    assert.equal(sender,
+      await tokenChannel.getSignerOfBalanceHash(20, senderSig)
+    );
+    const senderHash = await tokenChannel.generateKeccak256(senderSig);
+    const closingSig = signMsg(senderHash, privateKeys[receiver]);
+    await tokenChannel.uncooperativeClose(10, { from: sender });
+    await increaseTimeTo(latestTime() + 10);
+    await tokenChannel.cooperativeClose(20, senderSig, closingSig, { from: receiver });
+    (await token.balanceOf(sender)).should.be.bignumber
+      .equal(80);
+    (await token.balanceOf(receiver)).should.be.bignumber
+      .equal(20);
+  });
+});
```

Comment:
Please add tests/assertions for the following:
- Check that the contract is indeed destroyed after closing
- Trying to close the contract with a final balance greater than the collateral
- Sending incorrect/forged signatures in the cooperative close (signed with an incorrect private key, signed over a different balance, etc; for both the sender and the receiver signed messages)

Reply:
:+1: